Ignore style block and remove style tags from cue text.

eggei opened the pull request:
Thanks for sharing this tool. Helped a lot.

I just wanted to push an update to the parser because WEBVTT can have style block and tags. In my case, I have STYLE blocks and it raises `MalformedFile` error for my VTT.

Reference: https://www.w3.org/TR/webvtt1/#styling

Hopefully, someone is still checking the PRs ಠ‿ಠ

Cheers.

eggei commented:
Hey @brunoceleste, seems like you have given the go for the other PRs. Would you mind taking a look at this one please. If this doesn't work, please let me know and I will close the PR. Thank you!

brunoceleste commented:
Hi, thank you for your pull request!

Why would we want to strip the style from the text? I am not sure it should be the default behavior.
Any thought?

eggei commented:
Hi! - My thought was that without the style definition in context, there isn't a reason to keep the tags. Also, I thought it is the aim to keep `cue.text` as plain text and not a html-ish rich text.

If it was up to me, I might have created a new field called something like `cue.original_text` along with `cue.text` so both accessible.

eggei commented:
Hi @brunoceleste - Waiting for your directions to update or close this PR. Thank you!

brunoceleste commented:
Hi @eggei,

Sorry for the late reply. I like the idea of having `cue.original_text` to preserve style if present. Maybe rename it as `cue.raw_text` ? What do you think?

eggei commented:
@brunoceleste Sorry, it's been busy.

I updated this PR to create a new class attribute `@raw_text` which then I assign the original vtt text with style tags to it.
I added tests for this functionality as well as an addition to the README.

I wasn't sure about one thing. Let me know and I can update that too. -- Should `Cue#to_webvtt` use `@text` or `@raw_text`. I didn't touch it and it will just use the `@text`. Starting to use `@raw_text` might have been a breaking change for some. Or at least I thought so.

One last note, there is one false negative test on `master`.  Maybe you wanna check that.

eggei commented:
@brunoceleste Have a great week! Just reminding. No rush.

eggei commented:
Hello, @brunoceleste - sorry for the ping, just reminding.

brunoceleste commented:
Hi @eggei,

Sorry for the lack of response, been super busy lately :)

Good point on `#to_webvtt`. That's why I was a bit confused at first, because the style is removed by default. Why it's not the inverse, we keep it as raw text by default and add a field with just with text with no style. Either way, I think forcing a certain behavior feels not right and maybe it should be an option that the user gives to the parser function. Maybe it's better if he has the choice. What do you think? I am looking at the code right now.

eggei commented:
@brunoceleste Hmm, good point, okay, I think so. Maybe keeping the `@text` with all that is in there (including style tags) and having a _new_ attribute that provides the text _without_ any tags should be the strategy here. Sounds more straightforward as well. Any suggestion for this new attribute that'll hold the text without tags? `@plain_text`?

```rb
@text // string that appears in the VTT as is
@plain_text // string that is sanitized from style tags
```

brunoceleste commented:
`@plain_text` seems ok. We still use `@text` inside `#to_webvtt`, so no breaking change.
The problem is when we use `Blob#to_webvtt`, you will have all text, no plain text if you want to. We don't have parameters so we cannot just save the webvtt file without style.

I can merge with `@plain_text`, so you can access the cue without style if needed.

eggei commented:
Thank you @brunoceleste!
